Shared/Realm.Shared/RealmList.cs

```diff
@@ -316,6 +172,48 @@ private void AddObjectToRealmIfNeeded(T obj)
 
         #endregion
 
+        public override IEnumerator<T> GetEnumerator()
```

Comment:
Can we move this to `RealmCollectionBase` and make it shared between `List` and `Results`?